### PR TITLE
snap: support install/refresh that result in "Wait" changes in the cli better

### PR DESCRIPTION
**Pull request by mvo5**
On a classic system that supports mode the user experience when refreshing a snap in wait state (like the kernel) is not great. The progress meter just displays that a "Re-refresh" is in doing state. This is confusing for the user as really a reboot is required.

So this commit change the behavior so that after a refresh that ends up in a "Wait" snap will show a message:
```
# snap install --dangerous pc-kernel-new.snap 
2023-06-22T09:30:34Z INFO Task set to wait until a manual system restart allows
to continue
WARNING: pressing ctrl-c will abort the running change.
Change 18 waiting on external action to be completed

# snap changes 
...
18   Wait    today at 09:30 UTC  -                   Install "pc-kernel" snap from file "pc-kernel-new.snap"

# snap wait 18
error: the required argument `<key>` was not provided
root@localhost:/home/user1# snap watch 18
2023-06-22T09:30:34Z INFO Task set to wait until a manual system restart allows
to continue
Make snap "pc-kernel" (unset) available to the system
```

### CODE REVIEW

**Comment by pedronis** [comment]
@mvo5 should we remove the previous code that produced this:

WARNING: pressing ctrl-c will abort the running change.

?

**Review by pedronis — COMMENTED** [review]
this looks reasonable, as I asked I wonder which part of the last PR related to this we should now drop. Some small comments

File: cmd/snap/wait.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ type waitMixin struct {
 	clientMixin
 	NoWait    bool `long:"no-wait"`
 	skipAbort bool
+
+	// Wait also for tasks in the "wait" state.
+	waitWaitTasks bool
```
[the remark this answers is not in the document]
Comment (mvo5) [reply cmd/snap/wait.go:46]: renamed

File: cmd/snap/cmd_snap_op.go [hunk 2]
```diff
@@ -345,12 +345,17 @@ func maybeWithSudoSecurePath() bool {
...
+func showDone(cli *client.Client, chg *client.Change, names []string, op string, opts *client.SnapOptions, esc *escapes) error {
 	snaps, err := cli.List(names, nil)
 	if err != nil {
 		return err
 	}
 
+	if chg.Status == "Wait" {
+		fmt.Fprintf(Stdout, i18n.G("Change %v waiting on external action to be completed\n"), chg.ID)
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (mvo5) [reply cmd/snap/cmd_snap_op.go:352]: Uh, excellent point. I fixed this and added a unit test.

**Comment by mvo5** [comment]
> @mvo5 should we remove the previous code that produced this:
> 
> WARNING: pressing ctrl-c will abort the running change.
> 
> ?

Really sorry that I overlooked this, this is fixed now.

**Review by Meulengracht — APPROVED** [review]
This looks really nice! Thanks for doing this :-)

**Comment by pedronis** [comment]
after the last changes I think the description example output needs updating

**Review by pedronis — APPROVED** [review]
thanks, see my comment about the description though

**Pull request merged**